File: C++/Manual 07/Question02.cpp
#include <iostream>
using namespace std;

// Function to calculate grade based on marks
void calculateGrade(int marks, char &grade)
{
    if (marks > 80)
    {
        grade = 'A';
    }
    else if (marks > 65)
    {
        grade = 'B';
    }
    else if (marks > 50)
    {
        grade = 'C';
    }
    else
    {
        grade = 'F';
    }
}

int main()
{
    int marks;
    char grade;

    while (true)
    {
        cout << "Enter marks out of 100 (or -1 to exit): ";
        cin >> marks;

        if (marks == -1)
        {
            break;
        }

        calculateGrade(marks, grade);
        cout << "The grade is: " << grade << endl;
    }

    return 0;
}

File: Python/Manual 07/Question02.py
def calculateGrade(marks, grade):
    if marks > 80:
        grade[0] = "A"
    elif marks > 65:
        grade[0] = "B"
    elif marks > 50:
        grade[0] = "C"
    else:
        grade[0] = "F"


def main():
    while True:
        marks = int(input("Enter the marks (or enter -1 to exit): "))

        if marks == -1:
            break

        grade = [""]  # Using a list to simulate pass-by-reference
        calculateGrade(marks, grade)
        print("The grade is:", grade[0])


if __name__ == "__main__":
    main()
